internal/ingest/segmented_sync.go

```diff
@@ -0,0 +1,194 @@
+package ingest
```

Comment:
this filename doesn't describe the methods in it super well. not sure if there's a clearer naming convention between functions and filenames that might help

Reply:
Going with `synced.go` please shout if you can think of clearer alternatives